adapter/src/main/java/org/jvalue/ods/adapterservice/datasource/model/Datasource.java

```diff
@@ -39,6 +40,11 @@
   @NotNull
   private DatasourceTrigger trigger;
 
+  @OneToMany(cascade = CascadeType.ALL, mappedBy = "datasource")
+  @JsonIgnore
+  @EqualsAndHashCode.Exclude // needed to avoid an endless loop because of a circular reference
+  private Set<DataImport> dataImports;
```

Comment:
When are they fetched? Always on a get? Might lead to a huge overload once we had run 100 imports for a source... Maybe we should not link these concepts at all and just provide a datasourceId in the DataImport?

Reply:
I put in a `fetch = FetchType.LAZY` which should resolve your worries about performance.

> Maybe we should not link these concepts at all and just provide a datasourceId in the DataImport?

Usually I prefer things doing manually as well, I am not sure if all the Spring SQL magic still works then though. And to be honest, the foreign keys generated by using those annotations are kinda neat.

Reply:
I think, Hibernate only needs only of `@ManyToOne` or `@OneToMany` annotation to generate the foreign key.  Therefore just having the `@ManyToOne` in the DataImport class should be sufficient.

Reply:
I use that field to generate the result of the `/datasources/{id}/imports` call though. But of course that could also be done using a function in the repository.

Reply:
Yeah, I would definitely go with a separate `findAllByDatasourceId` function in the DataImportRespository. This will be much faster because querying all other properties of the Datasource is useless when just return the list of DataImports.

Reply:
Problem I see with this approach: 

We want to differentiate between "return empty array because no data imports were found for the data source" and "return 404 because datasource does not exist".

If we use a function called `findAllByDatasourceId` it would just return nothing if the datasource does not exist at all.
So we kinda need to fetch the datasource details anyways (or check if it exists, which should not be a big difference).

Reply:
Ah, I haven't considered that 🙈

Reply:
Good point.. I am kind of unsure what is best...

I guess lazy loading is misleading when reading code - or at least I'd assume the relation is fetched since I don't work with Spring too often. So personally, I'd not have the reference and make a separate call to prevent these uncertainties from happening. The readability would be clear: 

* if datasource does not exist => return 404
* if imports do not exist => return []

The obvious disadvantage: two database requests instead of one..

Reply:
> The obvious disadvantage: two database requests instead of one..

Well, I think having `@OneToMany` with `FetchType.LAZY` will also be two database requests...